[all] Refactor the loading of Xsp files.

damienmarchal opened the pull request:
On my way to fix this issue https://github.com/sofa-framework/sofa/issues/895
I found that the Xsp code was duplicated with a C and a C++ version. 

So I refactored the two and I added some tests to the files. 
On of the added test in ReadState_loadFailure is currently failing...wich is expected  as it shows an existing problem for which fixing is outside the scope of the PR. 
______________________________________________________
<!--- Please leave this at the end of your message -->
This PR: 
- [x] builds with SUCCESS for all platforms on the CI.
- [x] does not generate new warnings.
- [x] does not generate new unit test failures.
- [x] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [x] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

damienmarchal committed:
[SofaKernel] Deprecate MassSpringLoader in favor of a new class XspLoader

damienmarchal committed:
[SofaKernel] Deprecate SphereLoader API to unify it with XspLoader.

damienmarchal committed:
[SofaKernel] Update that was using MassSpringLoader & SphereLoader to use the new API

An exmaple of the API can be found in XspLoader/SphereLoader.

damienmarchal committed:
[SofaGeneralDeformable] Update VectorSpringForceField so it use the new XspLoader/SphereLoader API

damienmarchal committed:
[SofaGeneralLoader] Update MeshXspLoader so it use the new XspLoader/SphereLoader API

damienmarchal committed:
[SofaGeneralLoader_test] Refactor the ReadState_test and add a MeshXspLoader_test

damienmarchal committed:
[SofaKernel] Deprecate MassSpringLoader, use XspLoader instead.

damienmarchal committed:
[SofaKernel] Add XspLoader_test in framework_test.

damienmarchal commented:
I think the layout is ok now, 

I only want o re-implement the xsp loader using the code from MeshXspLoader that is stl instead of good old C. 

There is new tests including one that fails (because ...well the implementation is broken and the test is right ;))

guparan commented on SofaKernel/framework/framework_test/helper/io/XspLoader_test.cpp:
```diff
@@ -0,0 +1,133 @@
+/******************************************************************************
+*       SOFA, Simulation Open-Framework Architecture, development version     *
+*                (c) 2006-2018 INRIA, USTL, UJF, CNRS, MGH                    *
```
2019 :wink:

guparan commented on SofaKernel/framework/sofa/helper/CMakeLists.txt:
```diff
@@ -55,13 +55,14 @@ set(HEADER_FILES
     io/MeshTopologyLoader.h
     io/SphereLoader.h
     io/TriangleLoader.h
+    io/XspLoader.h
```
Already added line 58?

guparan commented on SofaKernel/framework/sofa/helper/io/MassSpringLoader.cpp:
```diff
@@ -47,7 +48,155 @@ static void skipToEOL(FILE* f)
+//    viscosity.endEdit();
+
+//    return true;
+//}
```
That's a big block of commented code here! :open_mouth:

damienmarchal replied:
Yes... this is the code made by Erik...in MeshXspLoader that I prefer to the "c" version that seems more complete.

Removing the C version by completing the Erik one is better.

guparan commented on SofaKernel/framework/sofa/helper/io/MassSpringLoader.h:
```diff
@@ -19,38 +19,8 @@
-#define SOFA_HELPER_IO_MASSSPRINGLOADER_H
 
-#include <sofa/defaulttype/Vec.h>
+XspLoader.has been deprecated and replaced with XspLoader.
```
**MassSpringLoader** has been deprecated :wink:

guparan commented on SofaKernel/framework/sofa/helper/io/XspLoader.cpp:
```diff
@@ -0,0 +1,400 @@
+/******************************************************************************
+*       SOFA, Simulation Open-Framework Architecture, development version     *
+*                (c) 2006-2018 INRIA, USTL, UJF, CNRS, MGH                    *
```
2019 :wink:

guparan commented on SofaKernel/framework/sofa/helper/io/XspLoader.cpp:
```diff
@@ -0,0 +1,400 @@
+//    viscosity.endEdit();
+
+//    return true;
+//}
```
Big block of commented code again.

guparan commented on SofaKernel/framework/sofa/helper/io/XspLoader.h:
```diff
@@ -0,0 +1,76 @@
+/******************************************************************************
+*       SOFA, Simulation Open-Framework Architecture, development version     *
+*                (c) 2006-2018 INRIA, USTL, UJF, CNRS, MGH                    *
```
2019 :wink:

guparan commented on modules/SofaGeneralLoader/SofaGeneralLoader_test/MeshXspLoader_test.cpp:
```diff
@@ -0,0 +1,95 @@
+/******************************************************************************
+*       SOFA, Simulation Open-Framework Architecture, development version     *
+*                (c) 2006-2018 INRIA, USTL, UJF, CNRS, MGH                    *
```
2019 :wink:

guparan reviewed (changes requested):
Here is a quick style review.

Could you add MassSpringLoader deprecation to ComponentChange.cpp ?

damienmarchal committed:
[SofaKernel] FIXUP MeshXspLoader.

damienmarchal committed:
[SofaKernel] Remove the C-implementation of XspLoader.

damienmarchal committed:
[SofaGeneralLoader] Update MeshXspLoader to use the new c++ implemetation.

damienmarchal committed:
Merge remote-tracking branch 'upstream/master' into fixIssue895

damienmarchal committed:
Merge remote-tracking branch 'upstream/master' into fixIssue895

damienmarchal committed:
[SofaKernel] Add new tests to XspLoader_test

damienmarchal committed:
[SofaKernel] Remove 'MassSpringLoader.cpp' is is now re-implemented in XspLoader.cpp

damienmarchal committed:
[SofaKernel] FIXUP XspLoader & MassSpringLoader refactoring.

damienmarchal committed:
[all] FIXUP related to XspLoader

damienmarchal commented:
@guparan I took into account your comments. 
I cannot add MassSpringLoader in ComponentChange because this is not a BaseObject.

damienmarchal commented:
[ci-build][with-scene-tests]

damienmarchal committed:
[SofaKernel] FIX XspLoader not reading the full line.

damienmarchal commented:
The ReadScene_test not passing is explicitly said so in the PR description, this is a new test showing an existing wrong behavior in Sofa (i.e. not sending an error message when trying to load an invalid faile).

[ci-build][with-scene-tests] will say if my last commit fixed the old scene failing

damienmarchal commented:
@sofa-framework/reviewers 

Is it possible to review this one before next week so it is on our back.

damienmarchal committed:
[SofaKernel] SphereLoader & XspLoader add doxygen.

Because I don't want to see on my grave that I'm commiting undocumented code :)

damienmarchal commented:
Ready...merge.
